dissect/target/filesystem.py

```diff
@@ -1238,6 +1232,44 @@ def map_file_entry(self, vfspath: str, entry: FilesystemEntry) -> None:
             entry_name = fsutil.basename(vfspath, alt_separator=self.alt_separator)
             directory.add(entry_name, entry)
 
+    def map_dir_from_tar(self, vfspath: str, tar_file: str | pathlib.Path, map_single_file: bool = False) -> None:
+        """Map files in a tar onto the VFS.
+
+        Args:
+            vfspath: Destination path in the virtual filesystem.
+            tar_file: Source path of the tar file to map.
+            map_single_file: Only mount a single file found inside the tar at the specified path.
+        """
+
+        if not isinstance(tar_file, pathlib.Path):
+            try:
+                tar_file = pathlib.Path(tar_file)
+            except TypeError:
+                raise ValueError("tar_file should be a string or Path instance")
+
+        vfspath = fsutil.normalize(vfspath, alt_separator=self.alt_separator)
+        tfs = TarFilesystem(tar_file.open("rb"))
+
+        if map_single_file:
+            # We map the first file we find in the tar to the provided vfspath.
+            for file in [f[0] for _, _, f in tfs.walk_ext("/") if f]:
+                file.name = fsutil.basename(vfspath)
+                self.map_file_entry(vfspath, file)
+                return
+
+        self.mount(vfspath, tfs)
```

Comment:
```suggestion
        if map_single_file:
            # We map the first file we find in the tar to the provided vfspath.
            for file in [f[0] for _, _, f in tfs.walk_ext("/") if f]:
                file.name = fsutil.basename(vfspath)
                self.map_file_entry(vfspath, file)
                return
        else:
            self.mount(vfspath, tfs)
```
Otherwise we may get an erroneous mount if you try to `map_single_file` with an empty tar. Harmless, but we can avoid it.